### DISCUSSION TITLE
Please provide a grammer for markdown format

### DISCUSSION THREAD

**Discussion by cdaman123** [ggml-org/llama.cpp#3678] [Ideas]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [X] I am running the latest code. Development is very rapid so there are no tagged versions as of now.
- [X] I carefully followed the [README.md](https://github.com/ggerganov/llama.cpp/blob/master/README.md).
- [X] I [searched using keywords relevant to my issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/filtering-and-searching-issues-and-pull-requests) to make sure that I am creating a new issue that is not already open (or closed).
- [X] I reviewed the [Discussions](https://github.com/ggerganov/llama.cpp/discussions), and have a new bug or useful enhancement to share.

# Expected Behavior

I am using `json_arr` grammer and when i asked for ordered list it gives me the comma seperated list without any heading.

# Current Behavior

I wanted a Markdown string which contain heading and ordered list inside a key of json structure. Like OpenAI function call give.

Thanks

**Comment by staviq** [comment]
[Why isn't there a formal grammar for Markdown?](https://roopc.net/posts/2014/markdown-cfg/)

**Comment by aman-gupta-doc** [comment]
@staviq Thanks for reply.

I go through the article. But we can write grammar for some elements of Markdown, like headings and lists (Ordered and Unordered).

Which will be helpful in answers.

**Comment by ejones** [comment]
Yeah my instinct would be to write an ad-hoc grammar for the task matching the particular Markdown structure desired. E.g.
```
./main -m ... -p '...' --grammar '
root   ::= header list
header ::= "# " [^\n]+ "\n\n"
list   ::= ("- " [^\n]+ "\n")+
'
